templates/components/engines/python/agent/tools/__init__.py

```diff
@@ -40,7 +31,7 @@ def load_tools(tool_type: str, tool_name: str, config: dict) -> list[FunctionToo
                 return tool_spec.to_tool_list()
             else:
                 module = importlib.import_module(f"{source_package}.{tool_name}")
-                tools = getattr(module, "tools")
+                tools = getattr(module, "get_tools")()
```

Comment:
basically, we just change the tool singleton instance into a function to initiate the tool per request.